### PR TITLE
domain reimplemented as class

### PR DESCRIPTION
**Pull request by d14b3l**
domain$Domain changed from type declaration to class definition as suggested in this issue:
https://github.com/facebook/flow/issues/1454

domain definition moved down so it would be able to extend eventEmmiter definition

### CODE REVIEW

File: lib/node.js [hunk 1]
```diff
@@ -499,9 +482,27 @@ declare class events$EventEmitter {
   setMaxListeners(n: number): void;
 }
 
+declare class EventEmitter extends events$EventEmitter {};
```
Comment (mroch) [inline lib/node.js]: this makes `EventEmitter` a valid global, which is not true is it? you have to import it from `events`?

would doing `declare class domain$Domain extends events$EventEmitter {` work?

Comment (d14b3l) [inline lib/node.js]: yes, it works fine, correction delivered

**Comment by mroch** [comment]
great, thanks!

now I'm wondering if `myDomain instanceof require('events').EventEmitter` would pass flow. I believe it's true at runtime. not sure how to handle this 100% accurately. I think we should merge this anyway since it's an improvement and instanceof is an edge case.

File: lib/node.js [hunk 2]
```diff
@@ -499,11 +482,29 @@ declare class events$EventEmitter {
   setMaxListeners(n: number): void;
 }
 
+
 declare module "events" {
   // TODO: See the comment above the events$EventEmitter declaration
   declare class EventEmitter extends events$EventEmitter {}
```
Comment (mroch) [inline lib/node.js]: can you change this to what you were changing it to before?

```
declare var EventEmitter : typeof events$EventEmitter;
```

that will fix the `instanceof` issue.

Comment (d14b3l) [inline lib/node.js]: Hi, correction delivered accordingly to your comment.

**Comment by mroch** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-3** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/545899388924725/int_phab) to review.

**Pull request closed without merging**